pkg/kubectl/service_basic.go

```diff
@@ -82,24 +83,38 @@ func (ServiceExternalNameGeneratorV1) ParamNames() []GeneratorParam {
 	}
 }
 
+// validatePortRange throws error if the given port is not vaild UDP or TCP port
+func validatePortRange(port int) error {
+	var err error
+	if port < 0 || port > 65535 {
+		err = errors.New("Invalid port")
+	}
+	return err
+}
+
 func parsePorts(portString string) (int32, intstr.IntOrString, error) {
 	portStringSlice := strings.Split(portString, ":")
 
 	port, err := strconv.Atoi(portStringSlice[0])
 	if err != nil {
 		return 0, intstr.FromInt(0), err
 	}
+	err = validatePortRange(port)
 	if len(portStringSlice) == 1 {
-		return int32(port), intstr.FromInt(int(port)), nil
+		return int32(port), intstr.FromInt(int(port)), err
 	}
 
 	var targetPort intstr.IntOrString
 	if portNum, err := strconv.Atoi(portStringSlice[1]); err != nil {
 		targetPort = intstr.FromString(portStringSlice[1])
 	} else {
+		err = validatePortRange(portNum)
+		if err != nil {
+			return 0, intstr.FromInt(0), err
+		}
 		targetPort = intstr.FromInt(portNum)
 	}
-	return int32(port), targetPort, nil
+	return int32(port), targetPort, err
```

Comment:
Should not return err here. As L108: `if portNum, err := strconv.Atoi(portStringSlice[1]); err != nil {` is a valid case, if you return err here, the caller will consider that `parsePorts` fails.